### PR TITLE
A37 update: add missing details

### PR DESCRIPTION
**Pull request by murgatroid99**
The specific changes are as follows:

 - Describe where the newly used data is in the `Cluster` proto message
 - Describe how errors from CDS and EDS watchers and DNS resolves should be handled
 - Describe the validation rules for the newly used data in the `Cluster` proto message
 - A couple of other notes from the conversation

### CODE REVIEW

File: A37-xds-aggregate-and-logical-dns-clusters.md [hunk 1]
```diff
@@ -95,7 +100,9 @@ For example:
 Each underlying cluster will have an associated discovery mechanism.
 The ordered list of discovery mechanisms will be passed down in the
 config for the new `xds_cluster_resolver_experimental` policy, which is the
-child of the CDS policy.
+child of the CDS policy. If an underlying cluster appears more than once in
+the list of clusters, only one discovery mechanism entry should be created for
+it.
```
Comment (markdroth) [inline A37-xds-aggregate-and-logical-dns-clusters.md:105]: Maybe add something like ", in the earliest position in the list in which it is seen"?

File: A37-xds-aggregate-and-logical-dns-clusters.md [hunk 2]
```diff
@@ -112,6 +119,11 @@ problem, we might be able to fix this later via subchannel pooling
 functionality).  However, we will share the atomic used for circuit breaking
 thresholds across all connections for a given cluster, as discussed below.
 
+If any of a CDS policy's watchers reports that the resource does not exist, the
```
Comment (markdroth) [inline A37-xds-aggregate-and-logical-dns-clusters.md:122]: Please also add a paragraph indicating that the tree of aggregate clusters must never be deeper than 16.  If the tree is deeper than that, then the CDS policy will report that it is in state TRANSIENT_FAILURE.

**Review by markdroth — COMMENTED** [review]
Thanks for doing this!  Just a couple of comments.

Please let me know if you have any questions.  Thanks!

File: A37-xds-aggregate-and-logical-dns-clusters.md (hunk 1 above)
Comment (murgatroid99) [reply A37-xds-aggregate-and-logical-dns-clusters.md]: Added

File: A37-xds-aggregate-and-logical-dns-clusters.md (hunk 2 above)
Comment (murgatroid99) [reply A37-xds-aggregate-and-logical-dns-clusters.md:125]: Added

**Review by markdroth — APPROVED** [review]

**Pull request merged**